app/mixins/item_actions.js

```diff
@@ -0,0 +1,157 @@
+import Ember from "ember";
+import AsyncMixin, { ERROR_STRATEGIES } from "./async";
+import _ from "lodash";
+
+/**
+ * Adds the following properties:
+ *
+ * - readyForAction (computed, true if other proerties are valid)
+ * - actionQty
+ * - actionTarget (the package to move)
+ * - actionFrom
+ * - actionName
+ * - editableQty
+ *
+ * Adds the following actions:
+ *
+ * - beginAction(package, from, action)
+ * - completeAction()
+ * - cancelAction()
+ */
+
+export default Ember.Mixin.create(AsyncMixin, {
+  locationService: Ember.inject.service(),
+  settings: Ember.inject.service(),
+  editableQty: Ember.computed.alias("settings.allowPartialOperations"),
+  actionComment: "",
+
+  itemActions: [
+    {
+      name: "Process",
+      icon: "random"
+    },
+    {
+      name: "Recycle",
+      icon: "recycle"
+    },
+    {
+      name: "Trash",
+      icon: "dumpster"
+    },
+    {
+      name: "Loss",
+      icon: "folder-minus"
+    }
+  ],
+
+  async resolveActionFromLocation(pkg) {
+    const presetLocations = pkg.get("packagesLocations").mapBy("location");
+
+    if (presetLocations.get("length") > 1) {
+      return this.get("locationService").userPickLocation({
+        headerText: this.get("i18n").t(
+          `select_location.${this.get("actionName").toLowerCase()}_from`
+        ),
+        presetLocations
+      });
+    }
+
+    return presetLocations.get("firstObject");
+  },
+
+  quantityAtLocation() {
+    const pkg = this.get("actionTarget");
+    const source = this.get("actionFrom");
+
+    if (!pkg) {
+      return 0;
+    }
+
+    const pkgLoc = pkg.get("packagesLocations").findBy("location", source);
+
+    return pkgLoc ? pkgLoc.get("quantity") : 0;
+  },
+
+  validActionParams() {
+    const target = this.get("actionTarget");
+    const from = this.get("actionFrom");
+    const actionName = this.get("actionName");
+    const qty = this.get("actionQty");
+
+    return target && from && actionName && qty > 0;
+  },
+
+  clearActionParams() {
+    this.set("readyForAction", false);
+    this.set("actionQty", 0);
+    this.set("actionTarget", null);
+    this.set("actionFrom", null);
+    this.set("actionName", null);
+    this.set("actionIcon", null);
+  },
+
+  actions: {
+    async beginAction(pkg, actionName) {
+      this.set("actionName", actionName);
+      let from = await this.resolveActionFromLocation(pkg);
+
+      if (!pkg || !actionName || !from) {
+        return this.send("cancelAction");
+      }
+
+      this.set(
```

Comment:
The backend _should_ prevent us from removing quantities that are designated to orders
but for a better experience we probably want to have `maxQuantity` set to

`Math.min([pkg.availableQuantity, quantityAtLocation])`